tests/test_types_typeddict.py

```diff
@@ -855,3 +864,17 @@ class Model(TypedDict):
     ta = TypeAdapter(Model)
 
     assert ta.validate_python({'x': 'ABC'}) == {'x': 'abc'}
+
+
+def test_schema_generator() -> None:
+    class LaxStrGenerator(GenerateSchema):
+        def str_schema(self) -> CoreSchema:
+            return core_schema.no_info_plain_validator_function(str)
```

Comment:
Not sure if this specifically is intended to be the solution for people who want to override the string validation, but if it is, we should make sure the JSON schema generation works. Right now, you get:

```
>>> ta.json_schema()
pydantic.errors.PydanticInvalidForJsonSchema: Cannot generate a JsonSchema for core_schema.PlainValidatorFunctionSchema ({'type': 'no-info', 'function': <class 'str'>})
```

Reply:
That'd be up to them to return a CoreSchema that can generate a JSON schema. I'm not sure there's much we can do.